src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -1040,9 +1041,20 @@ void DebugOnCmdStartDCmd::execute(DCmdSource source, TRAPS) {
 void CheckpointDCmd::execute(DCmdSource source, TRAPS) {
   Klass* k = SystemDictionary::resolve_or_fail(vmSymbols::jdk_crac_Core(),
                                                  true, CHECK);
-  JavaValue result(T_VOID);
+  JavaValue result(T_OBJECT);
+  JavaCallArguments args;
+  args.push_long((jlong )output());
+  args.push_long((jlong )LinuxAttachListener::get_jcmdOperation());
```

Comment:
You apparently don't need to pass the attach operation through java code to JVM_Checkpoint and eventually os::Linux::checkpoint -- you can get the operation right there with the same call.